### PR TITLE
IDSEQ-1795 - Synchronize auth0 cookies in the homepage

### PR DESCRIPTION
**Pull request by davidrissato**
# Description

Forces an auth0 user logout in the home page when users click login button

# Tests

* Manual tests using both strategies

### CODE REVIEW

File: app/controllers/auth0_controller.rb [hunk 1]
```diff
@@ -20,6 +20,10 @@ def refresh_token
     render :refresh_token, layout: false
   end
 
+  def login
+    redirect_to auth0_signout_url(url_for(action: :refresh_token, only_path: false, params: { mode: "login" }))
```
Comment (jshoe) [inline app/controllers/auth0_controller.rb]: nit: line 24 is a little hard to read. Might be cleaner on two lines?

File: app/assets/src/components/views/Landing.jsx [hunk 2]
```diff
@@ -78,7 +78,7 @@ class Landing extends React.Component {
   render() {
     const signInLink = () => {
       if (this.props.signInStrategy == "AUTH0") {
-        postToUrlWithCSRF("/auth/auth0");
+        location.href = "/auth0/login";
```
Comment (jshoe) [inline app/assets/src/components/views/Landing.jsx]: nice this will be much easier to use

File: app/controllers/auth0_controller.rb (hunk 1 above)
Comment (jshoe) [reply app/controllers/auth0_controller.rb]: Also maybe a comment on the mode 'login'? It seems unintuitive at first that the login action appears to go to `auth0_signout_url`

Comment (davidrissato) [reply app/controllers/auth0_controller.rb]: I just figured out a better approach to this problem. I don't need to invoke the sign_out URL anymore, I need to send a parameter `prompt=login` AND `connection=...` to the authorization endpoint.
It is worth pointing out that I had tried to send `prompt=login` before without any success, but it seems that `prompt=login` only works as expected if you send the connection name as well.
I'm making changes to this PR now.

**Pull request merged**